src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -132,17 +147,27 @@ Task PresentModal(Page modal, bool animated)
 			return source.Task.ContinueWith(task => NavAnimationInProgress = false);
 		}
 
+		void RestoreFocusability(AView nativeView)
+		{
+			nativeView.ImportantForAccessibility = ImportantForAccessibility.Auto;
+
+			if (NativeVersion.IsAtLeast(26))
+				nativeView.SetFocusable(ViewFocusability.FocusableAuto);
+
+			if (nativeView is ViewGroup vg)
+				vg.DescendantFocusability = DescendantFocusability.BeforeDescendants;
```

Comment:
Could you share an example so that I can understand this better? I'm wondering why it's [`.BeforeDescendants`](https://developer.android.com/reference/android/view/ViewGroup#FOCUS_BEFORE_DESCENDANTS) instead of [`.AfterDescendants`](https://developer.android.com/reference/android/view/ViewGroup#FOCUS_AFTER_DESCENDANTS). Is the `nativeview` here the rootView or the contained views?

Reply:
The container comes before the children in the hierarchy so by default TalkBack will consider the container first as a viable focus target before it moves onto the children. If we set it to after then the container would gain focus after navigating through the children. Which might be the desired result in some cases but it's not the default

the NativeView here is whatever view is currently going to be the visible one so it's just restoring the default value that it was set to before we blocked all focusability 

![image](https://user-images.githubusercontent.com/5375137/146431651-28300e10-4047-4b82-bc8f-00c3bd6f8abd.png)

Reply:
ah ok, makes sense! docs for `afterDescendants` was a big confusing and seemed right to me at first, since I was thinking the container should usually not receive any focus. But your explanation of "consider first as a viable focus target" helps clarify things